src/platform/GenericOTARequestorDriver.h

```diff
@@ -42,23 +44,45 @@ class GenericOTARequestorDriver : public OTARequestorDriver
         mImageProcessor = processor;
     }
 
+    // Set the timeout (in seconds) for the Default Provider timer
+    void SetDefaultProvidersTimeoutSec(uint32_t timeout) { mDefaultProvidersTimeoutSec = timeout; }
+
+    //// Virtual methods from OTARequestorDriver
     bool CanConsent() override;
     uint16_t GetMaxDownloadBlockSize() override;
-
     void HandleError(UpdateFailureState state, CHIP_ERROR error) override;
+    void HandleIdleState() override;
     void UpdateAvailable(const UpdateDescription & update, System::Clock::Seconds32 delay) override;
     void UpdateNotFound(UpdateNotFoundReason reason, System::Clock::Seconds32 delay) override;
     void UpdateDownloaded() override;
     void UpdateConfirmed(System::Clock::Seconds32 delay) override;
     void UpdateSuspended(System::Clock::Seconds32 delay) override;
     void UpdateDiscontinued() override;
     void UpdateCancelled() override;
+    void ScheduleDelayedAction(UpdateFailureState state, System::Clock::Seconds32 delay, System::TimerCompleteCallback action, void * aAppState) override;
+    void CancelDelayedAction(System::TimerCompleteCallback action, void * aAppState) override;
+    void ProcessAnnounceOTAProviders(const ProviderLocationType &providerLocation, 
+                                        app::Clusters::OtaSoftwareUpdateRequestor::OTAAnnouncementReason announcementReason) override;
+    void DriverSendQuery() override;
+    // Determines the next available Provider location and sets it in the OTARequestor
+    void DetermineAndSetProviderLocation() override;
+   
+    //// Regular methods
+    void StartDefaultProvidersTimer();
```

Comment:
Should these be private methods? Don't think they are used outside of this class?

Reply:
Fixed